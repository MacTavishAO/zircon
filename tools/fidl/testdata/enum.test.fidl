library test.name;

enum MyStrictEnum : uint32 {
    FOO = 1;
    BAR = 2;
};
