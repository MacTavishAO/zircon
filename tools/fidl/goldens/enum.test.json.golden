{
  "version": "0.0.1",
  "name": "test.name",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [
    {
      "name": "test.name/MyStrictEnum",
      "location": {
        "filename": "enum.test.fidl",
        "line": 3,
        "column": 6,
        "length": 12
      },
      "type": "uint32",
      "members": [
        {
          "name": "FOO",
          "location": {
            "filename": "enum.test.fidl",
            "line": 4,
            "column": 5,
            "length": 3
          },
          "value": {
            "kind": "literal",
            "value": "1",
            "expression": "1",
            "literal": {
              "kind": "numeric",
              "value": "1",
              "expression": "1"
            }
          }
        },
        {
          "name": "BAR",
          "location": {
            "filename": "enum.test.fidl",
            "line": 5,
            "column": 5,
            "length": 3
          },
          "value": {
            "kind": "literal",
            "value": "2",
            "expression": "2",
            "literal": {
              "kind": "numeric",
              "value": "2",
              "expression": "2"
            }
          }
        }
      ],
      "strict": true
    }
  ],
  "experimental_resource_declarations": [],
  "interface_declarations": [],
  "service_declarations": [],
  "struct_declarations": [],
  "table_declarations": [],
  "union_declarations": [],
  "type_alias_declarations": [],
  "declaration_order": [
    "test.name/MyStrictEnum"
  ],
  "declarations": {
    "test.name/MyStrictEnum": "enum"
  }
}
